c2/cuda/groth16_cuda.cu

```diff
@@ -111,6 +111,10 @@ RustError::by_value generate_groth16_proofs_c(const Assignment<fr_t> provers[],
     static std::mutex mtx;
     std::lock_guard<std::mutex> lock(mtx);
 
+    if (!ngpus()) {
+        return RustError{CUDA_ERROR_NO_DEVICE, "No CUDA devices available"};
```

Comment:
Just in case, **if** we change the signs, `ENODEV` would be better choice. Because the message would "your problem."